app/views/shipit/stacks/settings.html.erb

```diff
@@ -80,3 +90,11 @@
 
   </section>
 </div>
+<script type="text/javascript">
+function fetchCCMenuURL() {
+  $.ajax({url: "<%= ccmenu_url_url(stack_id: @stack.to_param) %>", success: function(result){
+    console.log(result);
+    $("#ccmenu-url").html("<input type='text' disabled value='" + result.ccmenu_url + "'/>");
```

Comment:
I meant the URL of the endpoint that this JS has to hit. `ccmenu_url_url(stack_id: @stack.to_param)` in this case.

Something like this:

```ruby
<%= link_to "Get CCMenu URL", ccmenu_url_url(stack_id: @stack.to_param), class: 'fetch-ccmenu-url' %>
```

```coffeescript
$(document).on 'click', '.fetch-ccmenu-url', (event) ->
  event.preventDefault()
  # Disable button
  $.get(event.target.href).done((token) ->
    $('#ccmenu-url').text(token).removeClass('hidden')
    # Remove button entirely
  ).fail(->
    # Reactivate the button somehow
  )
  
  
```